xformers/triton/k_softmax.py

```diff
@@ -71,7 +69,7 @@ def _softmax(
     if causal:
         io_mask = io_mask & (k <= n)
 
-    x = tl.load(x_ptrs, mask=io_mask, other=float("-inf"))
+    x = tl.load(x_ptrs, mask=io_mask, other=float("-inf")).to(tl.float32)
```

Comment:
doesn´t affect the IO, this is after the data has been loaded onto the SoC, removes subsequent casts (exp doesn´t accept bfloat16 for instance). Automatically cast back to the relevant type on the way out